### ISSUE TITLE
Support as python module

### ISSUE THREAD

**Issue by kecorbin** [imapex/boilerplate_sparkbot#6]
Filing this as enhancement, I wanted to use this work a little bit differently for a use case I was working on, so i have most of the work done to do something like the code sample below.  This would also enable distribution via pypi.

@hpreston @mbrainar  what do you think? i could probably get it PR'd into this repo pretty easy
  
```
import os
from bots.Spark import SparkBot

bot_email = os.getenv("SPARK_BOT_EMAIL")
spark_token = os.getenv("SPARK_BOT_TOKEN")
bot_url = os.getenv("SPARK_BOT_URL")
bot_app_name = os.getenv("SPARK_BOT_APP_NAME")

def do_something(incoming_msg):
    return "i did what you said - {}".format(incoming_msg.text)

bot = SparkBot(bot_app_name, spark_bot_token=spark_token,
               spark_bot_url=bot_url, spark_bot_email=bot_email)

bot.add_command('dosomething', 'help for do something', do_something)
bot.run(host='0.0.0.0', port=5000)
```

**Comment by hpreston** [comment]
@kecorbin  Yeah... I've been thinking that converting the boilerplate into a library/module like you're showing here would be a great way forward.  If you've got the code fleshed out, let's get it into a branch in this repo so we can kick the tires.  Once solid we can merge it into Master.  

Great work!

**Comment by kecorbin** [comment]
@chapeter @joshand @sluzynsk any feedback on the usage sample above?  Does that make sense?

Particularly the add_command method?

**Comment by sluzynsk** [comment]
Presumably a user would be then calling add_command multiple times, once for each bot command. 

I like it. It will hide all the implementation details from developers unless they *need* to see them. Several of the new teams this go were confused by all the code in the template that is ultimately just plumbing and not interesting to modify.

**Comment by kecorbin** [comment]
I'll have something up in a bit for you to test should we put ngrok in as an option as well?

Sent from my iPhone

On May 12, 2017, at 8:32 AM, Steve Luzynski <notifications@github.com<mailto:notifications@github.com>> wrote:


Presumably a user would be then calling add_command multiple times, once for each bot command.

I like it. It will hide all the implementation details from developers unless they need to see them. Several of the new teams this go were confused by all the code in the template that is ultimately just plumbing and not interesting to modify.

-
